Switch to dynssz codegen

pk910 opened the pull request:
This PR changes the code generator for ssz code to dynamic-ssz.

The code generated by dynamic-ssz has equal performance characteristics as the code generated by fastssz.

Biggest benefit of switching to the dynssz codegen:  no more pre- and post work to get the damn ssz code generated. The code generator just works out of the box 🎉. It uses types properly and does not require manual edits, which is a huge pain with the existing fastssz code generator.

supersedes #260, #242
fixes #259

pk910 committed:
dynssz annotation fixes for minimal preset compatibility

pk910 committed:
add missing `dynssz` annotations to api containers

pk910 committed:
make linter happy

pk910 committed:
re-generate blobsidecar_ssz.go

pk910 committed:
switch to dynamic-ssz codegen

pk910 committed:
generate legacy code only for direct comparison

pk910 committed:
bump `dynamic-ssz`

pk910 committed:
fix linter issues

pk910 committed:
flatten generated code

pk910 committed:
go fmt fixes

pk910 committed:
update generated code

pk910 committed:
regenerate code

pk910 committed:
switch to cli based code generator

pk910 committed:
bump dynamic-ssz & regen ssz code

pk910 committed:
re-generate ssz files

pk910 committed:
restore proof creation & verification methods

pk910 committed:
remove leftover codegen.go

pk910 committed:
regenerate code

pk910 committed:
reorder ssz code for easier comparison

pk910 committed:
regenerate code

pk910 committed:
regenerate ssz files

pk910 committed:
re-generate ssz code

pk910 committed:
re-generate ssz code

pk910 committed:
re-generate ssz files

pk910 committed:
Merge branch 'master' into pk910/dynssz-codegen

pk910 committed:
fix linter issues

pk910 committed:
bump `dynamic-ssz`

pk910 committed:
Merge branch 'master' into pk910/dynssz-annotation-fixes

pk910 committed:
Merge branch 'pk910/dynssz-annotation-fixes' into pk910/dynssz-codegen

pk910 committed:
update to dynamic-ssz v1.2.0

pk910 committed:
Merge branch 'master' into pk910/dynssz-codegen

pk910 committed:
fix golangci-lint issue

pk910 committed:
annotation fixes

pk910 committed:
regenerate leftover fastssz files

pk910 committed:
bump `dynamic-ssz` & regenerate ssz code

pk910 committed:
bump `dynamic-ssz` & re-generate ssz code

AntiD2ta committed:
Add Go version autoupdater workflow

AntiD2ta committed:
Add attgo-linter plugin to golangci-lint

AntiD2ta committed:
Remove invalid package-naming revive rule

The package-naming rule does not exist in revive as bundled with
golangci-lint v2.8.0 and causes config verify to fail.

pk910 committed:
Merge branch 'master' into pk910/dynssz-codegen

AntiD2ta committed:
Update attgo-linter to v0.2.0

pk910 committed:
fix `electra.BlindedBeaconBlockBody` annotation

pk910 committed:
fix linter issues

Bez625 commented on api/v1/blobs_ssz.go:
```diff
@@ -1,99 +1,72 @@
-				(*b)[ii] = new(deneb.Blob)
-			}
-			copy((*b)[ii][:], buf[ii*131072:(ii+1)*131072])
+func (t Blobs) UnmarshalSSZ(buf []byte) (err error) {
```
Is there a reason this has gone from a pointer receiver to a value receiver? I suspect this won't deserialise properly as we're

pk910 replied:
Yea, this is a bug that I've discovered recently too.
It affects lists only, structs are already passed as pointer receiver.
rel. https://github.com/pk910/dynamic-ssz/pull/147 (fixed as part of that PR)

Bez625 commented on api/v1/blobs_ssz.go:
```diff
@@ -1,99 +1,72 @@
 	return nil
 }
 
-// SizeSSZ returns the ssz encoded size in bytes for the Blobs object
```
Am I correct in thinking that dynamic-ssz does not include doc-strings? If not is there a plan to add them?

pk910 replied:
Yea, I've recently added some default doc strings:
https://github.com/pk910/dynamic-ssz/pull/145

Bez625 committed:
Non ssz files.

pk910 committed:
bump dynamic-ssz to v1.3.0

pk910 committed:
re-generate ssz code with v1.3.0

pk910 committed:
remove size tag from `uint256.Int` fields
the tag prevents using the the provided `uint256.Int.HashTreeRoot` function, because custom annotations cannot be passed down

pk910 commented:
@Bez625 
I've bumped the dynamic-ssz lib to release 1.3.0 and re-generated the ssz code.
This fixes both review comments :)

pk910 committed:
Merge remote-tracking branch 'upstream/master' into pk910/dynssz-codegen

pk910 committed:
bump dynamic-ssz

pk910 committed:
bump dynamic-ssz & move generator config into separate yaml files for maintainability

pk910 committed:
re-generate ssz files

pk910 commented:
@Bez625 
I've bumped the dynamic-ssz lib to release 1.3.1 and re-generated the ssz code (just semantic changes).

The 1.3.1 release mainly contains a new config format for the generator that simplifies the utterly long cli args in the generate.go
There's now a separate `generate.yaml` that lives along each `generate.go` and contains the types list in a more readable format.

Bez625 committed:
Update lib.

Bez625 committed:
Add dynssz.

Bez625 committed:
Address lint.

Bez625 committed:
Address review feedback.

Bez625 committed:
Merge pull request #291 from attestantio/dynssz-test

Switch fastssz to dynamic-ssz

pk910 committed:
Merge remote-tracking branch 'attestantio/master' into pk910/dynssz-codegen

pk910 committed:
bump dynamic-ssz & regen ssz code
